File: vocaflip-backend/src/main/java/com/vocaflipbackend/dto/response/StudySessionResponse.java
package com.vocaflipbackend.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;


//DTO phiên học Deck
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StudySessionResponse {
    private String id;
    private Integer totalCards;
    private Integer rememberedCount;
    private Integer forgotCount;
    private Integer durationSeconds;
    private LocalDateTime completedAt;
    private LocalDateTime createdAt;
    private String userId;
    private String deckId;

    // Danh sách thẻ cần học trong phiên này
    private List<StudyCardResponse> cards;
}

File: voca_flip_mobile/lib/data/models/responses/study_session_response.dart
import 'study_card_response.dart';

class StudySessionResponse {
  final String id;
  final int? totalCards;
  final int? rememberedCount;
  final int? forgotCount;
  final int? durationSeconds;
  final String? completedAt;
  final String? createdAt;
  final String? userId;
  final String? deckId;
  final List<StudyCardResponse> cards;

  const StudySessionResponse({
    required this.id,
    this.totalCards,
    this.rememberedCount,
    this.forgotCount,
    this.durationSeconds,
    this.completedAt,
    this.createdAt,
    this.userId,
    this.deckId,
    this.cards = const [],
  });

  /// JSON -> StudySessionResponse
  factory StudySessionResponse.fromJson(Map<String, dynamic> json) {
    return StudySessionResponse(
      id: json['id'] as String,
      totalCards: json['totalCards'] as int?,
      rememberedCount: json['rememberedCount'] as int?,
      forgotCount: json['forgotCount'] as int?,
      durationSeconds: json['durationSeconds'] as int?,
      completedAt: json['completedAt']?.toString(),
      createdAt: json['createdAt']?.toString(),
      userId: json['userId'] as String?,
      deckId: json['deckId'] as String?,
      cards:
          (json['cards'] as List<dynamic>?)
              ?.map(
                (e) => StudyCardResponse.fromJson(e as Map<String, dynamic>),
              )
              .toList() ??
          [],
    );
  }
}
